crypto/md5/md5_sha1.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015-2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <string.h>
+#include "internal/md5_sha1.h"
+#include <openssl/evp.h>
+
+int MD5_SHA1_Init(MD5_SHA1_CTX *mctx)
+{
+    if (!MD5_Init(&mctx->md5))
+        return 0;
+    return SHA1_Init(&mctx->sha1);
+}
+
+int MD5_SHA1_Update(MD5_SHA1_CTX *mctx, const void *data, size_t count)
+{
+    if (!MD5_Update(&mctx->md5, data, count))
+        return 0;
+    return SHA1_Update(&mctx->sha1, data, count);
+}
+
+int MD5_SHA1_Final(unsigned char *md, MD5_SHA1_CTX *mctx)
```

Comment:
I'd put the ctx param first. I know its how MD5_Final/SHA1_Final do it - but I don't think that's our style now.

Reply:
While I agree that the ctx being first is better, this is the signature being used by ALL digests, so it doesn't make sense to be inconsistent for this one case - especially when they are also being used inside a macro that assumes the params are in the (md, ctx) order.

Reply:
Ok